File: crates/facet_generate/src/generation/tests/with_enums_across_namespaces/snapshots/builtins_shadowed_across_namespaces/kotlin/com/example/kit/Kit.kt
package com.example.kit

import com.novi.bincode.BincodeDeserializer
import com.novi.bincode.BincodeSerializer
import com.novi.serde.DeserializationError
import com.novi.serde.Deserializer
import com.novi.serde.Serializer

fun <T> kotlin.collections.Set<T>.serialize(
    serializer: Serializer,
    serializeElement: Serializer.(T) -> Unit,
) {
    serializer.serialize_len(size.toLong())
    forEach { element ->
        serializer.serializeElement(element)
    }
}

fun <T> Deserializer.deserializeSetOf(deserializeElement: (Deserializer) -> T): kotlin.collections.Set<T> {
    val length = deserialize_len()
    val set = mutableSetOf<T>()
    repeat(length.toInt()) {
        set.add(deserializeElement(this))
    }
    return set
}

data class Set(
    val value: UInt,
) {
    fun serialize(serializer: Serializer) {
        serializer.increase_container_depth()
        serializer.serialize_u32(value)
        serializer.decrease_container_depth()
    }

    fun bincodeSerialize(): ByteArray {
        val serializer = BincodeSerializer()
        serialize(serializer)
        return serializer.get_bytes()
    }

    companion object {
        fun deserialize(deserializer: Deserializer): Set {
            deserializer.increase_container_depth()
            val value = deserializer.deserialize_u32()
            deserializer.decrease_container_depth()
            return Set(value)
        }

        @Throws(DeserializationError::class)
        fun bincodeDeserialize(input: ByteArray?): Set {
            if (input == null) {
                throw DeserializationError("Cannot deserialize null array")
            }
            val deserializer = BincodeDeserializer(input)
            val value = deserialize(deserializer)
            if (deserializer.get_buffer_offset() < input.size) {
                throw DeserializationError("Some input bytes were not read")
            }
            return value
        }
    }
}

data class Tray(
    val nothing: Unit,
    val ids: kotlin.collections.Set<UInt>,
) {
    fun serialize(serializer: Serializer) {
        serializer.increase_container_depth()
        serializer.serialize_unit(nothing)
        ids.serialize(serializer) {
            serializer.serialize_u32(it)
        }
        serializer.decrease_container_depth()
    }

    fun bincodeSerialize(): ByteArray {
        val serializer = BincodeSerializer()
        serialize(serializer)
        return serializer.get_bytes()
    }

    companion object {
        fun deserialize(deserializer: Deserializer): Tray {
            deserializer.increase_container_depth()
            val nothing = deserializer.deserialize_unit()
            val ids =
                deserializer.deserializeSetOf {
                    deserializer.deserialize_u32()
                }
            deserializer.decrease_container_depth()
            return Tray(nothing, ids)
        }

        @Throws(DeserializationError::class)
        fun bincodeDeserialize(input: ByteArray?): Tray {
            if (input == null) {
                throw DeserializationError("Cannot deserialize null array")
            }
            val deserializer = BincodeDeserializer(input)
            val value = deserialize(deserializer)
            if (deserializer.get_buffer_offset() < input.size) {
                throw DeserializationError("Some input bytes were not read")
            }
            return value
        }
    }
}

File: crates/facet_generate/src/generation/tests/with_enums_across_namespaces/snapshots/builtins_shadowed_across_namespaces/csharp/Example/kit/Kit.cs
using CommunityToolkit.Mvvm.ComponentModel;
using Facet.Runtime.Serde;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using Facet.Runtime.Bincode;

namespace Example.Kit;

public partial class Set : ObservableObject, IFacetSerializable, IFacetDeserializable<Set> {
    [ObservableProperty]
    private uint _value;

    public void Serialize(ISerializer serializer)
    {
        serializer.IncreaseContainerDepth();
        serializer.SerializeU32(Value);
        serializer.DecreaseContainerDepth();
    }

    public static Set Deserialize(IDeserializer deserializer)
    {
        deserializer.IncreaseContainerDepth();
        var value = deserializer.DeserializeU32();
        deserializer.DecreaseContainerDepth();
        return new Set {
            Value = value,
        };
    }

    public byte[] BincodeSerialize()
    {
        var serializer = new BincodeSerializer();
        Serialize(serializer);
        return serializer.GetBytes();
    }

    public static Set BincodeDeserialize(byte[] input)
    {
        if (input is null)
        {
            throw new DeserializationError("Cannot deserialize null array");
        }
        var deserializer = new BincodeDeserializer(input);
        var value = Deserialize(deserializer);
        if (deserializer.GetBufferOffset() < input.Length)
        {
            throw new DeserializationError("Some input bytes were not read");
        }
        return value;
    }
}

public partial class Tray : ObservableObject, IFacetSerializable, IFacetDeserializable<Tray> {
    [ObservableProperty]
    private global::Facet.Runtime.Serde.Unit _nothing;
    [ObservableProperty]
    private HashSet<uint> _ids;

    public void Serialize(ISerializer serializer)
    {
        serializer.IncreaseContainerDepth();
        serializer.SerializeUnit(Nothing);
        FacetHelpers.SerializeCollection(Ids, serializer, (item, s) => s.SerializeU32(item));
        serializer.DecreaseContainerDepth();
    }

    public static Tray Deserialize(IDeserializer deserializer)
    {
        deserializer.IncreaseContainerDepth();
        var nothing = deserializer.DeserializeUnit();
        var ids = FacetHelpers.DeserializeSet(deserializer, d => d.DeserializeU32());
        deserializer.DecreaseContainerDepth();
        return new Tray {
            Nothing = nothing,
            Ids = ids,
        };
    }

    public byte[] BincodeSerialize()
    {
        var serializer = new BincodeSerializer();
        Serialize(serializer);
        return serializer.GetBytes();
    }

    public static Tray BincodeDeserialize(byte[] input)
    {
        if (input is null)
        {
            throw new DeserializationError("Cannot deserialize null array");
        }
        var deserializer = new BincodeDeserializer(input);
        var value = Deserialize(deserializer);
        if (deserializer.GetBufferOffset() < input.Length)
        {
            throw new DeserializationError("Some input bytes were not read");
        }
        return value;
    }
}
